src/main/java/de/dhbw/karlsruhe/students/mailflow/external/infrastructure/cli/usecases/preferences/ChangePasswordCLIPrompt.java

```diff
@@ -0,0 +1,40 @@
+package de.dhbw.karlsruhe.students.mailflow.external.infrastructure.cli.usecases.preferences;
+
+import de.dhbw.karlsruhe.students.mailflow.core.application.auth.AuthSessionUseCase;
+import de.dhbw.karlsruhe.students.mailflow.core.application.usersettings.changepassword.ChangePasswordUseCase;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.auth.AuthorizationException;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.auth.HashingFailedException;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.auth.LoadingUsersException;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.user.exceptions.SaveUserException;
+import de.dhbw.karlsruhe.students.mailflow.external.infrastructure.cli.BaseCLIPrompt;
+
+public class ChangePasswordCLIPrompt extends BaseCLIPrompt {
+
+  private final ChangePasswordUseCase changePasswordUseCase;
+  private final AuthSessionUseCase authSession;
+
+
+  protected ChangePasswordCLIPrompt(BaseCLIPrompt previousPrompt,
+      ChangePasswordUseCase changePasswordUseCase, AuthSessionUseCase authSession) {
+    super(previousPrompt);
+    this.changePasswordUseCase = changePasswordUseCase;
+    this.authSession = authSession;
+  }
+
+  @Override
+  public void start() {
+    super.start();
+
+    String newPassword = simplePrompt("Enter new password:").trim();
+
+    try {
+      changePasswordUseCase.changePassword(authSession.getSessionUserAddress(), newPassword);
```

Comment:
The CLI should not know anything of the currently logged in user. This functionality is provided by the Application layer (see above comments)